nested-data-and-nested-loops/practice-nested-data-and-nested-loops.md

```diff
@@ -0,0 +1,403 @@
+    expected = "University of Washington"
+    self.assertEqual(name, expected)
+```
+##### !end-tests
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: b54addb3-48b0-462a-9f6a-f9b749fba39f
+* title: Practice Nested Data VI
+
+##### !question
+Modify `get_schools` to print the name and city of each school from `schools`.
+##### !end-question
+
+##### !placeholder
+```python
+schools = {
+    "school 1": {
+        "name": "Western Washington University",
+        "city": "Bellingham"
+    },
+    "school 2": {
+        "name": "University of Washington",
+        "city": "Seattle"
+    },
+    "school 3": {
+        "name": "Washington State University",
+        "city": "Pullman"
+    }
+}
+
+def get_schools():
+    pass
+```
+##### !end-placeholder
+##### !tests
+```python
+
+import main
+import unittest
+from unittest.mock import patch 
+
+@patch('builtins.print')
+class TestSchools(unittest.TestCase):
+  def test_schools(self, mock_print):
+    get_schools()
+    mock_print.assert_called_with("Pullman")
+```
+##### !end-tests
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: 279c4012-341e-4461-a716-f0c093c7faba
+* title: Practice Nested Data VII
+
+##### !question
+It's Karoake night! Ada's karoake machine has a list of performers and Akira is up next with their second song. Modify `get_song_name()` to return `Mr. Brightside` from the `performers` list.
+##### !end-question
+
+##### !placeholder
+```Python
+
+performers = {
+  "Rajan": ["I Want It That Way", "Jolene"],
+  "Akira": ["Return of the Mack", "Mr. Brightside"],
+  "Linh": ["Say My Name", "Since You Been Gone"]
+}
+
+def get_song_name():
+    pass
+```
+##### !end-placeholder
+##### !tests
+```python
+
+import main
+import unittest
+
+class TestSong(unittest.TestCase):
+  def test_song_name(self):
+    song = get_song_name()
+    expected = "Mr. Brightside"
+    self.assertEqual(name, expected)
+```
+##### !end-tests
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: b54addb3-48b0-462a-9f6a-f9b749fba39f
+* title: Practice Nested Data VIII
+
+##### !question
+Adie's want to keep record of all the songs performed during Karoake Night. Modify `get_songs()` to print all the songs from the `performers` list.
+##### !end-question
+
+##### !placeholder
+```Python
+
+performers = {
+  "Rajan": ["I Want It That Way", "Jolene"],
+  "Akira": ["Return of the Mack", "Mr. Brightside"],
+  "Linh": ["Say My Name", "Since You Been Gone"]
+}
+
+def get_songs():
+    pass
+```
+##### !end-placeholder
+##### !tests
+```python
+
+import main
+import unittest
+from unittest.mock import patch 
+
+@patch('builtins.print')
+class TestSchools(unittest.TestCase):
+  def test_schools(self, mock_print):
```

Comment:
Same feedback.  Consider multiple cases (use of `global`) and `StringIO` to capture results with full fidelity.